app/models/name_sorter.rb

```diff
@@ -129,15 +129,16 @@ def only_single_names
   end
 
   def push_synonym(arg)
-    if arg.is_a?(Integer)
+    case arg
```

Comment:
This is better than the if/elsif/else, but personally I'd do:
```
  def push_synonym(arg)
    return @approved_synonyms.push(Name.find(arg)) if arg.is_a?(Integer)
    return @approved_synonyms.push(arg) if arg.is_a?(ActiveRecord::Base)

    raise(TypeError.new(
            "NameSorter synonyms must be Integer or ActiveRecord::Base, "\
            "not #{arg.class}."
          ))
  end
```

Reply:
Yes, that's better.